### ISSUE TITLE
Missing deprecation guides

### ISSUE THREAD

**Issue by locks** [ember-learn/deprecation-app#893]
- [ ] Ember.Component.reopen `ember.component.reopen`[Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/9609b8f30e0cdbc2235063f571af619f4362f73b/packages/@ember/-internals/glimmer/lib/component.ts#L1156)
- [ ] Passing tagName to LinkTo `ember.link-to.tag-name` [Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/57907a4d7d91aa58b3b3ac1258e639ce221f9492/packages/@ember/-internals/glimmer/lib/components/link-to.ts#L432)
- [ ] Passing disabledWhen to LinkTo `ember.link-to.disabled-when` [Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/57907a4d7d91aa58b3b3ac1258e639ce221f9492/packages/@ember/-internals/glimmer/lib/components/link-to.ts#L627) + 1 more in the same file
- [ ] Old deprecate method imports `old-deprecate-method-paths` [Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/22bfcfdac0aeefcf333fb2d6697772934201b43b/packages/@ember/application/de precations.ts#L8) + 1 other location in the same file
- [ ] Deprecate without `for` `ember-source-deprecation-without-for` [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/4984045459bc9f5f34f1ae8b51fc6522fb72cbf4/packages/@ember/debug/lib/deprecate.ts#L211) 
- [ ] Deprecate without `since` `ember-source-deprecation-without-since` [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/4984045459bc9f5f34f1ae8b51fc6522fb72cbf4/packages/@ember/debug/lib/deprecate.ts#L224) 
- [ ] EXTEND_PROTOTYPES `ember-env.old-extend-prototypes` [Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/566af4941eb7ecaa4679e295ecfc38263d5ee1fe/packages/ember/index.js#L175)
- [ ] Class-based template compilation plugins `template-compiler.registerPlugin` [Deprecation Guide]() [Deprecation Invocation Site](https://github.com/emberjs/ember.js/blob/e28a201b9ccfb16f93089cea9207fca9b552d98c/packages/ember-template-compiler/lib/system/compile-options.ts#L93)

**Comment by ef4** [comment]
I think also `deprecated-run-loop-and-computed-dot-access`.

**Comment by chancancode** [comment]
That’s correct. In practice it can probably share most of the prose with that globals deprecation, you mostly experience that as a special case of the globals access deprecation

**Comment by Glarregle** [comment]
I will work on the following: 

**Ember.Component.reopen**
**Passing tagName to LinkTo**
**Passing disabledWhen to LinkTo**
**Old deprecate method imports**

**Comment by ijlee2** [comment]
@Glarregle Awesome, thank you!

**Comment by jenweber** [comment]
@Glarregle I discovered that the tagName deprecation already exists: https://deprecations.emberjs.com/v3.x/#toc_ember-built-in-components-legacy-arguments so this one does not need a new writeup.

**Comment by Glarregle** [comment]
@jenweber I noticed that `old-deprecate-method-paths`  deprecation is gone from [source code](https://github.com/emberjs/ember.js/tree/master/packages/%40ember/application) already

**Comment by bartocc** [comment]
Hi, @fannyvanrell and myself would like to work on `EXTEND_PROTOTYPES` if that's still needed.

cc: @monsieurdart @pbernery

**Comment by ijlee2** [comment]
@bartocc @fannyvanrell Thank you, I think we do need help with providing details for https://deprecations.emberjs.com/v3.x#toc_ember-env-old-extend-prototypes.

Can you also fix the typo in the `h3` header? `protptypes` should read `prototypes`.

<img width="640" alt="Old extend prototypes" src="https://user-images.githubusercontent.com/16869656/134191437-94126f4d-a0b9-49b4-a7a6-e075dfbf568d.png">

**Comment by bartocc** [comment]
Hi again. We'd like to go for `Deprecate without for` and `Deprecate without since` if that's OK.
cc @fannyvanrell
